src/components/filter/Filter.tsx

```diff
@@ -91,59 +64,82 @@ export const Filter = ({ filterTypes, initialFilters, setQueryVariables }: Props
 
-      <Collapsible collapsed={isCollapsed}>
-        <WrapperVertical style={styles.noPaddingTop}>
-          {filterTypes.map((item) => (
-            <WrapperVertical key={item.name} style={styles.noPaddingBottom}>
-              {item.type === FILTER_TYPES.DATE && (
-                <DateFilter
-                  filters={filters}
-                  setFilters={setFilters}
-                  containerStyle={{ width: device.width * 0.45 }}
-                  {...item}
-                  data={item.data as { name: keyof FilterProps; placeholder: string }[]}
-                />
-              )}
-
-              {item.type === FILTER_TYPES.DROPDOWN && item.data?.length && (
-                <DropdownFilter
-                  filters={filters}
-                  setFilters={setFilters}
-                  {...item}
-                  data={item.data as DropdownProps[]}
-                />
-              )}
-
-              {item.type === FILTER_TYPES.SUE.STATUS && (
-                <StatusFilter
-                  filters={filters}
-                  setFilters={setFilters}
-                  {...item}
-                  data={item.data as StatusProps[]}
-                />
-              )}
-            </WrapperVertical>
-          ))}
-        </WrapperVertical>
-
-        <WrapperVertical style={styles.noPaddingBottom}>
-          <Button
-            invert
-            disabled={isNoFilterSet}
-            title={texts.filter.resetFilter}
-            onPress={() => {
-              setIsCollapsed(true);
-
-              setTimeout(() => {
-                setFilters(initialFilters);
-              }, 500);
+      {isOverlay ? (
+        <Modal
+          animationType="slide"
+          onRequestClose={() => setIsCollapsed(!isCollapsed)}
+          presentationStyle="pageSheet"
+          visible={!isCollapsed}
+        >
+          <Header
+            backgroundColor={colors.transparent}
+            centerComponent={{
+              text: texts.filter.filter,
+              style: { color: colors.darkText, fontSize: normalize(18), fontWeight: '700' }
+            }}
+            rightComponent={{
+              icon: 'close',
+              color: colors.darkText,
+              onPress: () => setIsCollapsed(!isCollapsed)
             }}
           />
-        </WrapperVertical>
-
-        <Divider />
-      </Collapsible>
-    </>
+          <Divider />
+          <Wrapper style={styles.noPaddingBottom}>
+            <FilterComponent filters={filters} filterTypes={filterTypes} setFilters={setFilters} />
+          </Wrapper>
+
+          <WrapperVertical style={styles.noPaddingTop}>
+            <WrapperRow center spaceAround>
+              <Button
+                disabled={!!isNoFilterSet}
+                invert
+                notFullWidth
+                onPress={() => {
+                  setIsCollapsed(!isCollapsed);
+
+                  setTimeout(() => {
+                    setFilters(initialFilters);
```

Comment:
testing the reset, the slider value remains as is selected it before

Reply:
it's still the case, right? the slider values is not being reset:

![Simulator Screen Recording - iPhone 15 Plus - 2024-09-27 at 16 40 29](https://github.com/user-attachments/assets/fc54c37c-c8fb-4450-979e-d6bfed57cfd8)

Reply:
I solved the problem.